### PR TITLE
Add missing @Override annotations to CliGitAPIImpl

### PR DESCRIPTION
**Pull request by darxriggs**
This makes it consistent with `JGitAPIImpl`.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java [hunk 1]
```diff
@@ -2789,8 +2834,8 @@ public void deleteBranch(String name) throws GitException, InterruptedException
 
     }
 
-
```
Comment (MarkEWaite) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: Please don't cleanup whitespace as part of a pull request.  It creates many more merge conflicts for pending pull requests (including those closed and labeled with 'Later')

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java [hunk 2]
```diff
@@ -2912,15 +2959,14 @@ public void execute() throws GitException, InterruptedException {
         };
     }
 
-
```
Comment (MarkEWaite) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: Please don't cleanup whitespace as part of a pull request.  Merge conflicts increase dramatically with white space changes

Comment (darxriggs) [reply src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: fixed

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java (hunk 1 above)
Comment (darxriggs) [reply src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: fixed

**Pull request merged**